src/devices/BoardLed/BoardLed.cs

```diff
@@ -0,0 +1,170 @@
+        public IEnumerable<string> EnumerateTriggers()
+        {
+            _triggerReader.DiscardBufferedData();
+            _triggerReader.BaseStream.Position = 0;
+
+            return Regex.Replace(_triggerReader.ReadToEnd(), @"\[|\]", "")      // remove selected chars
+                .Split(' ');
+        }
+
+        /// <summary>
+        /// Get all BoardLed instances of on-board LEDs.
+        /// </summary>
+        /// <returns>BoardLed instances.</returns>
+        public static IEnumerable<BoardLed> EnumerateLeds()
+        {
+            var infos = Directory.GetDirectories(DefaultDevicePath)
+                .Where(x => !x.Contains(':'))       // remove items like "mmc0::"
+                .Select(x => new DirectoryInfo(x));
+
+            return infos.Select(x => new BoardLed(x.Name));
+        }
+
+        /// <summary>
+        /// Cleanup
+        /// </summary>
+        public void Dispose()
+        {
+            _brightnessReader?.Dispose();
+            _brightnessWriter?.Dispose();    
+            _triggerReader?.Dispose();
+            _triggerWriter?.Dispose();
+            _maxBrightnessReader?.Dispose();
+
+            _brightnessReader = null;
+            _brightnessWriter = null;
+            _triggerReader = null;
+            _triggerWriter = null;
+            _maxBrightnessReader = null;
+        }
+
+        private int GetBrightness()
+        {
+            _brightnessReader.DiscardBufferedData();
+            _brightnessReader.BaseStream.Position = 0;
+
+            return int.Parse(_brightnessReader.ReadToEnd());
+        }
+
+        private int GetMaxBrightness()
+        {
+            _maxBrightnessReader.DiscardBufferedData();
+            _maxBrightnessReader.BaseStream.Position = 0;
+
+            return int.Parse(_maxBrightnessReader.ReadToEnd());
+        }
+
+        private void SetBrightness(int value)
+        {
+            if (value > 255)
+            {
+                value = 255;
+            }
+            else if (value < 0)
+            {
+                value = 0;
+            }
+
+            _brightnessWriter.BaseStream.Position = 0;
+
+            _brightnessWriter.Write(value);
+            _brightnessWriter.Flush();
+        }
+
+        private string GetTrigger()
+        {
+            _triggerReader.DiscardBufferedData();
+            _triggerReader.BaseStream.Position = 0;
+
+            return Regex.Match(_triggerReader.ReadToEnd(), @"(?<=\[)(.*)(?=\])").Value;
+        }
+
+        private void SetTrigger(string name)
+        {
+            _triggerWriter.BaseStream.Position = 0;
+
+            _triggerWriter.Write(name);
+            _triggerWriter.Flush();
+        }
+
+        private void Initialize()
+        {
+            FileStream brightnessStream = File.Open($"{DefaultDevicePath}/{Name}/brightness", FileMode.Open);
+            _brightnessReader = new StreamReader(brightnessStream, Encoding.ASCII, false, 2, true);
```

Comment:
use named args and if there is some buffering passed use the smallest possible or no buffering option if possible